voltixApp/VoltixApp/Chains/Polkadot.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  Polkadot.swift
+//  VoltixApp
+//
+
+import Foundation
+import Tss
+import WalletCore
+import BigInt
+
+enum PolkadotHelper {
+    
+    static let defaultFeeInPlancks: BigInt = 200000000 //0.02
+    
+    static func getPolkadot(hexPubKey: String, hexChainCode: String) -> Result<Coin, Error> {
+        return getAddressFromPublicKey(hexPubKey: hexPubKey, hexChainCode: hexChainCode).flatMap { addr -> Result<Coin, Error> in
+            TokensStore.createNewCoinInstance(ticker: "DOT", address: addr, hexPublicKey: hexPubKey, coinType: .polkadot)
+        }
+    }
+    
+    static func getAddressFromPublicKey(hexPubKey: String, hexChainCode: String) -> Result<String, Error> {
+        // Polkadot is using EdDSA , so it doesn't need to use HD derive
+        guard let pubKeyData = Data(hexString: hexPubKey) else {
+            return .failure(HelperError.runtimeError("public key: \(hexPubKey) is invalid"))
+        }
+        guard let publicKey = PublicKey(data: pubKeyData, type: .ed25519) else {
+            return .failure(HelperError.runtimeError("public key: \(hexPubKey) is invalid"))
+        }
+        return .success(CoinType.polkadot.deriveAddressFromPublicKey(publicKey: publicKey))
+    }
+    
+    static func getPreSignedInputData(keysignPayload: KeysignPayload) -> Result<Data, Error> {
+        guard keysignPayload.coin.chain.ticker == "DOT" else {
+            return .failure(HelperError.runtimeError("coin is not DOT"))
+        }
+        
+        guard case .Polkadot(let recentBlockHash, let nonce, let currentBlockNumber, let specVersion, let transactionVersion) = keysignPayload.chainSpecific else {
+            return .failure(HelperError.runtimeError("fail to get to address"))
+        }
+        guard let toAddress = AnyAddress(string: keysignPayload.toAddress, coin: .polkadot) else {
+            return .failure(HelperError.runtimeError("fail to get to address"))
+        }
+        
+        let genesisHash = Data(hexString: "0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3")!
```

Comment:
Instead of hardcode the genesisHash , can you please get it by using the RPC call,  it seems you can get it by calling `chain_genesis`